shared-module/canio/Message.c

```diff
@@ -28,16 +28,14 @@
 
 #include <string.h>
 
-void common_hal_canio_message_construct(canio_message_obj_t *self, int id, void *data, size_t size, bool rtr, bool extended)
+void common_hal_canio_message_construct(canio_message_obj_t *self, int id, void *data, size_t size, bool extended)
 {
     self->id = id;
     self->size = size;
-    self->rtr = rtr;
+    self->rtr = !data;
```

Comment:
Please just split Message and RTR fully. It's easier understand a consistent structure.